### PR TITLE
Reworks space ninjas to be less overpowered and obnoxious, adds new space ninja gamemode.

### PR DESCRIPTION
**Pull request by WhyIsCaeciliusTaken**
## About The Pull Request

Currently, space ninjas are less of a ninja and more of a "hyper-buffed teleporting DEsword nukeop on meth". They're absurdly powerful and not very fun to fight, and don't really feel like ninjas at all, for several reasons:

1: They're too fast. Permanent meth speed plus the ability to teleport through walls makes hitting them extremely difficult. They're easily the most mobile antagonist by far, and combined with their massive burst damage in the form of shock gloves + katana or throwing star spam makes them almost impossible to kill unless you have armoury grade weapons or the ninja is dumb enough to let you baton and cuff them. You can't even steal their katana because they can recall it unlimited times for little to no cost.

2: They deal too much damage. Between the instant fight ender stun gloves, the katana that does 40 damage and the ultra spammable throwing stars, ninjas kill people absurdly quickly. This, combined with the high speed, means most encounters with a ninja result in you being blindsided by the speed, stun gloved then killed while you're stunned with no chance to fight back. A ninja is supposed to be stealthy, not run in the front door and kill everyone.

3: They have way too many escape methods. Smoke, adrenals, teleports, meth speed. Each on their own is powerful, but combined means that if a ninja doesn't want to fight you, you can do more or less nothing about it. Ninjas kill you absurdly quickly when they want to fight, and you can't do anything to them back as they can just run away spamming their escape tools.

4: Their objectives encourage total round removal. Ninja nets are essentially like cremating someone if they aren't freed in time, as they teleport you to the holding facility for the rest of the round. NT sided ninjas are worse, as they're encouraged by their objectives to specifically hunt down and kill antagonists.

In conclusion, ninjas are horribly imbalanced and not fun for anyone but the ninja, that specifically makes the game less fun for other antagonists in the round. As such, I have taken steps to make them more focused on sneaking around and being an actual ninja rather than just killing everyone in sight.

Ninjas now move at normal speed, and spawn with a choice beacon and no katana. Most of the special features of their suit, their spammable escape tools, nets and ability to drain APCs have been removed, and the armour has been slightly lowered. The katana recall ability has been limited to three uses, and moved to their gloves, which can now fabricate a limited number of gadgets to help them in combat: bolas, smoke grenades and a new item, caltrops, which puncture people's feet when walked over, damaging and stunning them, even through shoes. Their choice beacon allows them to select one of five "paths", which are specialisations containing additional weapons and equipment. The default path, the path of balance, contains the following:

- An energy katana without the ability to teleport
- A cloaker belt. This item allows ninjas to become invisible for a brief time. It recharges while turned on in dark areas.
- 2 Ninja Stars, still allowing you to use them but not spam them.
- An adrenal implant

The other selectable paths are:

Speed:
- Dash Katana (old energy katana with teleports, but a longer teleport cooldown)
- A cloaker belt
- Adrenal implant
- No flash proof NVGs

Might:
- Mighty Energy Katana, an improved version with better blocking abilities. It can block ~6-8 bullets before causing you to drop the katana with stamcrit.
- M.I.G.H.T. suit, an up armoured suit for head on confrontations.
- A shorter duration cloaker belt.

Silence:
- a SILENCE cloaker belt, with a much longer duration.
- 1 sleepy pen
- A ninja baton instead of a katana. Knocks down, does heavy stamina damage and mutes the target on hit. Can also silently emag objects on hit on harm intent.

Wisdom:
- No weapon at all. This is probably by far the hardest to use.
- Standard Cloaker Belt
- Bluespace Emag (emag that works from a distance)
- SMART ninja gloves. To make up for the lack of weapon, the SMART gloves grant Krav Maga, have additional fabrication charges and can produce ninja stars as well as stronger versions of the items the normal gloves can (wide AoE smokes, reinforced bolas and grenades that scatter caltrops everywhere).

Ninjas now all work for the syndicate, and get a syndicate comms key, allowing possible teamplay with other syndicate agents. Due to ninjas being weaker overall, their assassination objectives are easier and are no longer only tasked to kill antagonists or heads of staff.

Also adds a gamemode called "Ninja Incursion" which pits the station against multiple ninjas at once trying to complete different objectives. It might end up having to be admin only but we can at least give it a go and see how it pans out. After all, it's quite different to most other modes, being a stealth antag that relies on darkness and cloaking to remain unseen rather than disguises.

Additionally, ninjas are now more common to compensate for them being weaker.

## Why It's Good For The Game

Old ninjas suck for everyone except the ninja, are terribly coded, and aren't even really a ninja at all. These new ninjas are more like an actual ninja, aren't a horrible mess code wise and will hopefully be more fun to play against.
## Changelog
:cl:
add: Space ninjas work completely differently now. They're no longer super fast death machines and must rely on stealth, their new cloaking belt and gadgets to get by.
add: Added caltrops and caltrop grenades, new tools for ninjas. They briefly stun and damage the legs of anyone who walks over them, even with shoes on.
add: Ninjas can now choose one of five "paths", each with unique sets of gear.
add: Ninja Incursion gamemode, where the station is assaulted by multiple ninjas at once
del: Removed ninja nets, most of the old ninja suit features such as EMP pulses.
tweak: Space ninja events can trigger earlier now and are more common
balance: default energy katana can no longer teleport. There is a new, more powerful version that still can.
balance: reduced space ninja movement speed to standard.
balance: space ninjas now have less armour unless they take the might path.
balance: recall energy katana now has a max of three uses.
balance: space ninjas now have a cloaking belt, which allows them to become invisible for a short period. Recharges while turned on in darkness.
balance: ninja star fabrication and stun recovery are now unique to certain paths. All ninjas can now fabricate smoke bombs, caltrops and ninja energy bolas.
imageadd: added caltrop icon
/:cl:

### CODE REVIEW

**Comment by Goblinu** [comment]
this seems great, honestly i every fight with the ninja is over in a instant

**Comment by OceanFish1** [comment]
Dunno but every round ninja appears on the station a few minutes later someone says in comms "Ninja is ded". With this OP gear they still somehow manage to die.

**Comment by Singularity69** [comment]
ninja is intended to be A mobile antagonist though? why nerf their mobility

**Comment by ghost** [comment]
doesnt this mean you could just meme ninja with A stun baton/disabler because now they cant escape easily?

**Comment by jlsnow301** [comment]
Good riddance old ninja... Objective screen was like "1. Nothing - Success!"

**Comment by PowerfulBacon** [comment]
damn I like the really OP ninja, watching the assistants try to validhunt and getting absolutely steamrolled is a good feeling.

From when I played ninja before you can absolutely destroy the greytides however security was a bit harder to deal with since you can't really take them on due to the abilities costing a ridiculous amount of power.
(Requires about 3 APC charges to get to full power then after EMPing twice you are completely out of charge and have to go charge again)
The main issue was with how easy it was to get away, I was going into to security killing 1 or 2 guards, then as soon as they got me to near death I just teleported into space and went to maints to heal up (Although healing isn't their strongpoint since the suit takes a minute to take off and then another minute to put back on again).

**Comment by WhyIsCaeciliusTaken** [comment]
> Dunno but every round ninja appears on the station a few minutes later someone says in comms "Ninja is ded". With this OP gear they still somehow manage to die.

Most ninjas are not very smart. They tend to just flat out not use half their weapons so you either get a bad ninja that dies in under 5 minutes or a good ninja who just kills everyone with little effort/counterplay.



> doesnt this mean you could just meme ninja with A stun baton/disabler because now they cant escape easily?

If they run up to you like an idiot yes. They still have escape tools here (cloak, smoke, caltrops), it's just that none of them are "I'm gone now and you can't do jack" like the old ninja.

When I think of space ninja, I think of "stealthy dude who sneaks around", not "hotline miami".

**Comment by PowerfulBacon** [comment]
the speed was the best part of it :(
Also what did the adrenals even do before? I popped all of them at once (not realising they are limited) and it didn't even protect me from stuns.
I liked the shadow operative thing when it was played, but at the same time I like ninjas too and thought they should be seperate

**Comment by WhyIsCaeciliusTaken** [comment]
> the speed was the best part of it :(
> Also what did the adrenals even do before? I popped all of them at once (not realising they are limited) and it didn't even protect me from stuns.

Reset stuns that you currently had on you and added radiation to you. They didn't protect from stuns applied after you used the button.

File: code/game/gamemodes/ninja/ninja.dm [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+	enemy_minimum_age = 14
+
+	announce_span = "danger"
+	announce_text = "The station is being infiltrated by spider clan operatives!\n\
+	<span class='danger'>Ninjas</span>: Remain undetected and complete your objectives!\n\
+	<span class='notice'>Crew</span>: Locate and kill the ninjas!"
+
+/datum/game_mode/ninja/pre_setup()
+	recommended_enemies = round(num_players() / 10)
+	var/datum/mind/ninja = antag_pick(antag_candidates, ROLE_NINJA)
```
Comment (PowerfulBacon) [inline code/game/gamemodes/ninja/ninja.dm]: wont this just spawn 1 ninja always?

**Comment by Sinmero** [comment]
The paths seem to be poorly balanced. People can tell its ninja just by looking at them and with no teleport katana they are very restricted on station access.
Speed paths seems like an obvious choice and Might/Balance are ok but Silence and Wizdom seem bad.
Sleepy pen is good but takes a white to take effect and krav maga is not that great after the nerf.
I think you should add Chameleon to silence path armor and for wizdom you should add apc drain to replenish charges and emp.
Also since paths with no katana are limited in access it might be a good idea to add ability to pry open airlocks to gloves (like xenos or jaws do)

**Comment by eeSPee** [comment]
I like the idea of ninja incursion, and I get that ninja is a super OP powerhouse, but I think these guys are really weak and can't do their basic tasks anymore. Here are some concerns that didn't matter as much for ninja, but might be a big issue for these guys:

* Stealth is not as strong and AI could trace them. They should have digital camo by default (like changeling ability)
* Without an energy katana teleport in or emag doors, there is no stleathy way to get in the station. Most outside airlocks are bolted and taking arrivals is too risky. 
* Katana smacking is super loud and may reveal them for simple things such as navigating maints. This may not have been an issue for the old ninjas, but this can be a problem for current ninjas. A stealthy emag would be nice, in place of the katana emagging stuff.
* 1 TC traitor kit contains more than 2 ninja stars.
* You seem to forget that ninjas get validhunted like mad when exposed/heard. The original ninja could get out of this with the katana, which made the validhunting not as worth it. These guys? Not so much.

>Speed

Seems the only one worthwhile, IMO. The lack of teleport katanas puts the other ones to a very large disadvantage. Seems that this is the only path that can punish validhunters and escape sec.

>Might

For a stealth antag that gets validhunted, is there really a reason to get this. The suit is not that robust and the katana is just speed katana without the teleport. What prevents this one from getting stunned, flashbanged or simply being shoved against a wall and having his katana stolen? Or just harass him at range with stamina weapons, 2 shots and he's down. None of his abilities will help him. Not really well thought out.

>Silence

Sleepy pen on its own is not that strong (30 chloral, 15 morphine?) and without refilling, this is gonna be pointless. An endlessly refilling one would be nice. This one seems really underwhelming. No substitute for katana/teleport makes this ninja unable to enter the station.

>Wisdom

Krav maga not that good. Has the same problems silence has when navigating the station. 

Also give them japanese names instead of SMART, SILENCE, MIGHT, etc.

**Comment by WhyIsCaeciliusTaken** [comment]
> I like the idea of ninja incursion, and I get that ninja is a super OP powerhouse, but I think these guys are really weak and can't do their basic tasks anymore. Here are some concerns that didn't matter as much for ninja, but might be a big issue for these guys:
> 
> * Stealth is not as strong and AI could trace them. They should have digital camo by default (like changeling ability)
> * Without an energy katana teleport in or emag doors, there is no stleathy way to get in the station. Most outside airlocks are bolted and taking arrivals is too risky.
> * Katana smacking is super loud and may reveal them for simple things such as navigating maints. This may not have been an issue for the old ninjas, but this can be a problem for current ninjas. A stealthy emag would be nice, in place of the katana emagging stuff.
> * 1 TC traitor kit contains more than 2 ninja stars.
> * You seem to forget that ninjas get validhunted like mad when exposed/heard. The original ninja could get out of this with the katana, which made the validhunting not as worth it. These guys? Not so much.
> 
> > Speed
> 
> Seems the only one worthwhile, IMO. The lack of teleport katanas puts the other ones to a very large disadvantage. Seems that this is the only path that can punish validhunters and escape sec.
> 
> > Might
> 
> For a stealth antag that gets validhunted, is there really a reason to get this. The suit is not that robust and the katana is just speed katana without the teleport. What prevents this one from getting stunned, flashbanged or simply being shoved against a wall and having his katana stolen? Or just harass him at range with stamina weapons, 2 shots and he's down. None of his abilities will help him. Not really well thought out.
> 
> > Silence
> 
> Sleepy pen on its own is not that strong (30 chloral, 15 morphine?) and without refilling, this is gonna be pointless. An endlessly refilling one would be nice. This one seems really underwhelming. No substitute for katana/teleport makes this ninja unable to enter the station.
> 
> > Wisdom
> 
> Krav maga not that good. Has the same problems silence has when navigating the station.
> 
> Also give them japanese names instead of SMART, SILENCE, MIGHT, etc.

These are all valid concerns, and to be honest the reason SILENCE and WISDOM are so bad is because they were going to have an additional item they could make which turned out to be completely broken overpowered (smokebombs which knocked people out who entered it). Additionally, ninjas are untrackable by the AI by default because their hood acts like an agent card, and the ninja stars in the 1TC kit are the weaker lower damage version. The other concerns are still very valid and I'll get to tweaking them now.

It should also be noted that the cloaking belt is very powerful, especially in dark areas.

**Comment by Sinestia** [comment]
raging ninjas pog

**Comment by eeSPee** [comment]
>/datum/game_mode/ninja/pre_setup()
>	recommended_enemies = round(num_players() / 10)
>	if (prob(3))
>		recommended_enemies = 999

Noballs

**Comment by WhyIsCaeciliusTaken** [comment]
Balance changes:

Buffed all paths except speed and might. Rebalanced might (overall a buff).

Balance gains an adrenalin implanter and improved, flashproof NVGs.  This will hopefully make them a jack of all trades, being able to do a little bit of what each of the paths can do, but not as well as the specialised ones.

Might has been focused more on combat. It also gains the flashproof NVGs, along with an improved katana that has a significantly higher block power and class, allowing them to block more powerful attacks on a wider arc. However, their cloak duration has been shortened, reinforcing them as the "Loud" path.

Silence gains a new weapon, the ninja baton. It is retractable and similar to the contractor baton. Additionally, it mutes the target for a short duration on hit, and can silently emag items it hits if it is on harm intent and extended.

Wisdom can now produce ninja stars with its gloves, and gains a bluespace emag, allowing it to emag objects from afar.

**Comment by Sinestia** [comment]
> Dunno but every round ninja appears on the station a few minutes later someone says in comms "Ninja is ded". With this OP gear they still somehow manage to die.

Player skill is still a factor

even nightmares who are much weaker than ninjas can wind up killing like half of the station when played by someone with skill against people without. Doesn't mean the average doesn't just break like 2 apcs and get deaded.

 Ninjas are so rare and the average player is pretty unrobust so no wonder it's usually like that.

**Comment by WhyIsCaeciliusTaken** [comment]
This is ready for review, feel free to roast my code.

**Comment by MNarath1** [comment]
I actually played ninja today and while it has a lot of op abilities to get away its kinda needed to do not get insta roasted by sec most of your advanced abilities rely on energy that you have to get from apc's thats also kinda interresting and while sec wasn't able to kill me they heavily injured me and all of them got out of the net i tried to catch them in for my objective its meant to be quite a strong antag but it still can be beaten trough not by a single person but when crew actually works together

**Comment by MNarath1** [comment]
not gonna lie most of this looks like a little bit better traitor

**Comment by MNarath1** [comment]
trough maybe we could include it as a lesser ninja also wtf is the dme file for old ninja like 20 dm files beeing loaded why is it spread between so much files

**Comment by Fronsis** [comment]
Really like the idea of the rework i'd personally would like to still seeing a bit of speed on them from time to time, maybe make their adrenals charge a little faster than average adrenal? 

I agree that they still need a way to teleport/get across the station but without being a spammable jail free card

Obviously getting dragged into the space dojo for the rest of the round sucks but the place is quite fancy and i wouldn't like to see it go, i think we should still use that in someway or another, maybe keep their kidnap objective but not something like 6 living specimens maybe something like three or it could be kinda like the contractor when certain treshold is met they're released back to the station but brainwashed so they can't scream I GOT ABDUCTED BY NINJA! and reveal them, i'd rather prefer a stealthy ninja with some in-out ways than the one we currently have that's just SPEEEEEED

**Comment by bloons3** [comment]
>Obviously getting dragged into the space dojo for the rest of the round sucks but the place is quite fancy and i wouldn't like to see it go

It's a cool place for all of 5 minutes, and then it's just round removal.

**Comment by ItsTisme** [comment]
I always feel like cloak is pretty weak especially for sec who gets sec huds that very cleary mark even a invisible ninja. And as everyone mentiond without emag/tp katana they are way to limited in mobility especialy if they cant keep invis for long times.

Considering these will be super obvs valids they need alot better equipment to actuely survive the assault from everyone

Also sleepy pen doesnt work on ninjas, Are they gonna sneak into chemistry to refill the pen after its 1 use?

**Comment by WhyIsCaeciliusTaken** [comment]
> I always feel like cloak is pretty weak especially for sec who gets sec huds that very cleary mark even a invisible ninja. And as everyone mentiond without emag/tp katana they are way to limited in mobility especialy if they cant keep invis for long times.
> 
> Considering these will be super obvs valids they need alot better equipment to actuely survive the assault from everyone
> 
> Also sleepy pen doesnt work on ninjas, Are they gonna sneak into chemistry to refill the pen after its 1 use?

This cloak hides you from sechuds I'm pretty sure. I'd need to test it though. They do get emags, and agent cards.

**Comment by eeSPee** [comment]
You should really look at warframe for ideas.

Like, 'Excalibur' could be your generic dashing, star throwing ninja, but you could also have stuff like the 'Mag' warframe which throws, pulls and knocks enemies around, or the 'Banshee' one that has strong sound attacks (think aoe flashbangs), 'Vauban' for electric/tesla attacks, etc.

**Comment by utilbird** [comment]
> You should really look at warframe for ideas.

Lootbox PR coming soon

File: code/modules/mob/living/carbon/human/human.dm [hunk 2]
```diff
@@ -104,16 +68,16 @@
...
+		else
 			tab_data["Tank Pressure"] = GENERATE_STAT_TEXT("[internal.air_contents.return_pressure()]")
 			tab_data["Distribution Pressure"] = GENERATE_STAT_TEXT("[internal.distribute_pressure]")
 
 	if(mind)
 		var/datum/antagonist/changeling/changeling = mind.has_antag_datum(/datum/antagonist/changeling)
 		if(changeling)
-			tab_data["Chemical Storage"] = GENERATE_STAT_TEXT("[changeling.chem_charges]/[changeling.chem_storage]")
 			tab_data["Absorbed DNA"] = GENERATE_STAT_TEXT("[changeling.absorbedcount]")
+			tab_data["Chemical Storage"] = GENERATE_STAT_TEXT("[changeling.chem_charges]/[changeling.chem_storage]")
```
Comment (park66665) [inline code/modules/mob/living/carbon/human/human.dm]: I think there were meaningless line shufflings happened here.

File: code/modules/antagonists/ninja/suit/suit.dm [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+// contains all of the ninja clothes except the gloves, because they're complex enough to warrant their own file.
+
+/obj/item/clothing/suit/space/space_ninja
+	name = "ninja suit"
+	desc = "A unique, vacuum-proof suit of nano-enhanced armor designed specifically for Spider Clan assassins."
+	icon_state = "s-ninja"
+	item_state = "s-ninja_suit"
+	allowed = list(/obj/item/gun, /obj/item/ammo_box, /obj/item/ammo_casing, /obj/item/melee/baton, /obj/item/restraints/handcuffs, /obj/item/tank/internals, /obj/item/stock_parts/cell)
+	slowdown = 0
+	resistance_flags = LAVA_PROOF | ACID_PROOF
```
Comment (park66665) [inline code/modules/antagonists/ninja/suit/suit.dm]:
```suggestion
	resistance_flags = LAVA_PROOF | FIRE_PROOF | ACID_PROOF
```

I think you meant this...?

File: code/modules/antagonists/ninja/suit/n_glove_verbs/ninja_fabricate.dm [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+/obj/item/clothing/gloves/space_ninja/proc/ninjafabricate(var/mob/living/carbon/human/user)
+	if(fabrication_charges > 0)
+		var/choice = input(user,"Which item would you like to fabricate?","Select an Item") as null|anything in sortList(fabrication_options)
+		var/obj/new_item = new choice()
+		if(user.put_in_hands(new_item))
+			to_chat(user, "<span class='notice'>A [new_item] has been created in your hand! The gloves have [fabrication_charges] left.</span>")
+			fabrication_charges--
+		else
+			qdel(choice)
+		user.throw_mode_on() //So they can quickly throw it.
```
Comment (park66665) [inline code/modules/antagonists/ninja/suit/n_glove_verbs/ninja_fabricate.dm]:
```suggestion
			fabrication_charges--
			user.throw_mode_on() //So they can quickly throw it.
		else
			qdel(choice)
```

You also can add warning text for not having a free hand.

File: code/modules/antagonists/ninja/suit/gloves.dm [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+/obj/item/clothing/gloves/space_ninja/wisdom
+	name = "smart ninja gloves"
+	desc = "Advanced ninja gloves with a wider variety of better quality fabrications. Also teaches the user Krav Maga."
+	fabrication_charges = 15
+	fabrication_options = list(/obj/item/restraints/legcuffs/bola/tactical = "heavy bola", /obj/item/grenade/clusterbuster/smoke = "large smoke bomb", /obj/item/grenade/clusterbuster/caltrops = "caltrop scatter grenade", /obj/item/throwing_star/ninja = "throwing star")
+	var/datum/martial_art/krav_maga/style = new
+
+/obj/item/clothing/gloves/space_ninja/wisdom/equipped(mob/user, slot)
+	if(!ishuman(user))
```
Comment (park66665) [inline code/modules/antagonists/ninja/suit/gloves.dm]:
```suggestion
/obj/item/clothing/gloves/space_ninja/wisdom/equipped(mob/user, slot)
	..()
	if(!ishuman(user))
```

Note: if you really want to disable katana recall ability for smart ninja gloves, you should override `actions_list`

File: code/modules/antagonists/ninja/suit/gloves.dm [hunk 6]
```diff
@@ -0,0 +1,53 @@
...
+		return
+	if(slot == SLOT_GLOVES)
+		var/mob/living/carbon/human/H = user
+		style.teach(H,1)
+
+/obj/item/clothing/gloves/space_ninja/wisdom/dropped(mob/user)
+	if(!ishuman(user))
```
Comment (park66665) [inline code/modules/antagonists/ninja/suit/gloves.dm:50]:
```suggestion
/obj/item/clothing/gloves/space_ninja/wisdom/dropped(mob/user)
	..()
	if(!ishuman(user))
```

File: code/game/gamemodes/ninja/ninja.dm [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+	for(var/obj/effect/landmark/carpspawn/S in GLOB.landmarks_list)
+		if(isturf(S.loc))
+			spawn_locs += S.loc
+		if(!spawn_locs.len)
+			return
+	var/spawn_loc = pick(spawn_locs)
+	if(!spawn_loc)
+		return MAP_ERROR
+	for(var/datum/mind/ninjanew in ninjas)
+		ninjanew.current.forceMove(spawn_loc)
```
Comment (park66665) [inline code/game/gamemodes/ninja/ninja.dm]: Ninja spawn location handling should be done with `/datum/antagonist/ninja/proc/movetospawn`, which is called by `/datum/antagonist/ninja/greet`, which is called by `/datum/antagonist/proc/on_gain`, which is called by `/datum/antagonist/ninja/on_gain`, which is called by `ninja.add_antag_datum(/datum/antagonist/ninja)` line below.
```suggestion
```

File: code/game/gamemodes/ninja/ninja.dm (hunk 1 above)
Comment (park66665) [reply code/game/gamemodes/ninja/ninja.dm]: Just wrap ninja picking and adding lines with a for loop.

File: code/modules/mob/living/carbon/human/human.dm [hunk 8]
```diff
@@ -60,42 +60,6 @@
 		tabs.Insert(1, "SpiderOS")
 	return tabs
```
Comment (park66665) [inline code/modules/mob/living/carbon/human/human.dm]: This part is only for SpiderOS... which you got rid of. You should remove this override too.

File: code/modules/antagonists/ninja/gear.dm [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+
+/obj/item/melee/classic_baton/contractor_baton/ninja
+	name = "ninja baton"
+	desc = "A retractable baton. Mutes targets on hit. Can also override doors."
+
+/obj/item/melee/classic_baton/contractor_baton/ninja/additional_effects_carbon(mob/living/target, mob/living/user)
+	. = ..()
+	if(iscarbon(target))
+		var/mob/living/carbon/M = target
+		M.silent += 10
```
Comment (park66665) [inline code/modules/antagonists/ninja/gear.dm]:
```suggestion
	var/mob/living/carbon/C = target
	C.silent += 10
```

This proc is `additional_effects_carbon`, after all.

File: code/modules/antagonists/ninja/gear.dm [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+/obj/item/storage/box/syndi_kit/ninja
+	name = "path of balance"
+
+/obj/item/storage/box/syndi_kit/ninja/PopulateContents()
+	new /obj/item/energy_katana(src)
+	new /obj/item/shadowcloak/ninja(src)
+	new /obj/item/clothing/glasses/night/ninja(src)
+	new /obj/item/throwing_star/ninja(src)
+	new /obj/item/throwing_star/ninja(src)
+	new /obj/item/implanter/adrenalin(src)
```
Comment (park66665) [inline code/modules/antagonists/ninja/gear.dm:68]: I don't think `/obj/item/implanter/adrenalin` should be included to the path of balance.

File: code/modules/antagonists/ninja/gear.dm [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+	new /obj/item/pen/sleepy(src)
+	new /obj/item/melee/classic_baton/contractor_baton/ninja(src)
+
+/obj/item/storage/box/syndi_kit/ninja/wisdom
+	name = "path of wisdom"
+
+/obj/item/storage/box/syndi_kit/ninja/wisdom/PopulateContents()
+	new /obj/item/shadowcloak/ninja(src)
+	new /obj/item/clothing/gloves/space_ninja/wisdom(src)
+	new /obj/item/card/emag/bluespace(src)
```
Comment (park66665) [inline code/modules/antagonists/ninja/gear.dm:101]: I have two questions.

1. I though NVS was a rather common component. Were NVGs omitted accidently?
2. I didn't hear about bluespace emags. Is there anything wrong?

File: code/modules/antagonists/ninja/ninja.dm [hunk 12]
```diff
@@ -36,11 +32,7 @@
 	for(var/datum/mind/M in SSticker.minds)
 		if(M.current && M.current.stat != DEAD)
 			if(ishuman(M.current))
-				if(M.special_role)
-					possible_targets[M] = 0						//bad-guy
-				else if(M.assigned_role in GLOB.command_positions)
-					possible_targets[M] = 1						//good-guy
-
+				possible_targets[M] = 0
```
Comment (park66665) [inline code/modules/antagonists/ninja/ninja.dm]:
```suggestion
				possible_targets += M
```

I don't think dictionaries are necessary now. We can store possible targets to a non-assoc list.

File: code/datums/actions/ninja.dm [hunk 13]
```diff
@@ -1,51 +1,15 @@
 /datum/action/item_action/initialize_ninja_suit
 	name = "Toggle ninja suit"
```
Comment (park66665) [inline code/datums/actions/ninja.dm]:
```suggestion
```

I am pretty sure this item action is now obsolete.

**Review by park66665 — COMMENTED** [review]
I ~~finally~~ got around taking a cursory look at the PR. Feel free to ask.

File: code/modules/mob/living/carbon/human/human.dm (hunk 2 above)
Comment (WhyIsCaeciliusTaken) [reply code/modules/mob/living/carbon/human/human.dm]: Yea that's unintentional. I had to fix some merge conflicts due to TGUI stat panel and removing the SpiderOS parts.

File: code/modules/antagonists/ninja/suit/n_glove_verbs/ninja_fabricate.dm (hunk 4 above)
Comment (WhyIsCaeciliusTaken) [reply code/modules/antagonists/ninja/suit/n_glove_verbs/ninja_fabricate.dm]: Done.

File: code/modules/antagonists/ninja/gear.dm (hunk 11 above)
Comment (WhyIsCaeciliusTaken) [reply code/modules/antagonists/ninja/gear.dm:101]: All ninjas get regular NVGs in their outfit on spawn. the /ninja subtype is an improved version that also gives flash protection. Come to think of it the ninja hood also grants flash protection, so it's a bit pointless. I'll remove it.

Bluespace emags are intentional, I just forgot to add it to the PR description. It's just an emag that works from a distance, fits the "Gadgets" theme of the path.

File: code/game/gamemodes/ninja/ninja.dm (hunk 7 above)
Comment (WhyIsCaeciliusTaken) [reply code/game/gamemodes/ninja/ninja.dm]: Fixed

File: code/modules/antagonists/ninja/gear.dm (hunk 10 above)
Comment (WhyIsCaeciliusTaken) [reply code/modules/antagonists/ninja/gear.dm:68]: I disagree. Without it, why wouldn't you take Might instead for the deflection on the katana and better armour? If you really want to I'll remove it but I'm not sure why it's an issue.

Comment (park66665) [reply code/modules/antagonists/ninja/gear.dm:68]: The only reason I objected was it wasn't included in the PR description.

**Comment by WhyIsCaeciliusTaken** [comment]
I've tried my best to do balancing, but if this is to be properly balanced, I'd need a testmerge to see how the crew fight ninjas. After all, stealth antags that rely on not being seen at all are not very common, and I'd need to see how the crew fight them.

For example: I think the combination of cloak, smoke, darkness and dropping caltrops behind them will be enough to defeat stun baton cheese, but I can't really test that properly by myself.

**Comment by DatBoiTim** [comment]
I like ninja incursion as an idea, with some tweaks. It's approaching the issue of Wizard, so rare it's hard to learn, and since it's only 1 person, a ninja event/round doesn't give multiple people a chance simultaneously. I think maybe consider breaking up some of the ninja abilities to the paths in addition to what they have(where applicable)

But please keep it separate from the classic ninjas, since they will get fucked by this(though meth speed removal sounds ideal)

**Comment by WhyIsCaeciliusTaken** [comment]
No it doesn't. My master is up to date with current and it merges fine. What gives?

**Comment by PowerfulBacon** [comment]
I think they should stay a midround and not have a gamemode for it

**Comment by ItsTisme** [comment]
> I think they should stay a midround and not have a gamemode for it

I think a problem is if there arent enough ghosts to pull a large enough team, it would fuck over the spawned team

**Comment by PowerfulBacon** [comment]
I mean they shouldnt be a team but solo

**Comment by ItsTisme** [comment]
> I mean they shouldnt be a team but solo

Arent they a little weak to be solo? I thought the whole point was that they where going to be teams

**Comment by PowerfulBacon** [comment]
They are incredibly overpowered with full invisbility and a 2 hit kill weapon, they wiped out an entire nukie team. I would say they are even more overpowered before, because previously the abilities like invisibility were not spammable and had limited uses or energy costs.

And the nukies didn't even put up a fight

**Comment by ItsTisme** [comment]
> They are incredibly overpowered with full invisbility and a 2 hit kill weapon, they wiped out an entire nukie team. I would say they are even more overpowered before, because previously the abilities like invisibility were not spammable and had limited uses or energy costs.
> 
> And the nukies didn't even put up a fight

A man thats immune to agro grab killed a intire team of CQC users yes, not very comparable

**Comment by PowerfulBacon** [comment]
Anyway here is the feedback
 - Being full invisible during combat isn't that fun. Its good for stealth but not good while the combat is going on.
 - They should not have syndicate comms, after all they are a spider clan ninja.
 - They shouldn't be a team or a gamemode and should be a solo midround antagonist.
 - The abilities of the ninja should be linked to the ninja in someway and not useable by the rest of the crew.

**Comment by ItsTisme** [comment]
>The abilities of the ninja should be linked to the ninja in someway and not useable by the rest of the crew.
They should not have syndicate comms, after all they are a spider clan ninja.
Being full invisible during combat isn't that fun. Its good for stealth but not good while the combat is going on.

agreed, maby after there first strike it goes on a little cooldown?

**Comment by MNarath1** [comment]
https://github.com/tgstation/tgstation/pull/53492 a possible alternative of completely gibbing old ninja

**Comment by amidoingitright** [comment]
get rid of everything else, remove their EMP and make EMP's absolutly shred space ninja, they are after all just a super augmented human

**Comment by FortSumter** [comment]
The path names aren't weeb enough.

**Comment by ike709** [comment]
Stale

**Pull request closed without merging**